components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/password/AdminForcePasswordResetManager.java

```diff
@@ -39,22 +42,31 @@ public static AdminForcePasswordResetManager getInstance() {
      *
      * @return auto-generated one time password
      */
-    public String generatePassode() {
-        return IdentityUtils.generatePasscode(6);
+    public String generatePasscode() {
+        return IdentityUtils.getInstance().generatePasscode(6);
     }
 
     /**
      * pass the otp to the store to persist it in the database
      *
      * @param userUniqueId selected user id
-     * @param passCode     generated code
      * @throws IdentityRecoveryException
      */
-    public void persistPasscode(String userUniqueId, String passCode) throws IdentityRecoveryException {
+    public String persistPasscode(String userUniqueId) throws IdentityRecoveryException {
         UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
-        UserRecoveryData recoveryDataDO = new UserRecoveryData(userUniqueId, passCode,
+        String passcode = generatePasscode();
+        if (log.isDebugEnabled()) {
+            log.debug("The passcode is generated.");
```

Comment:
Add more details to the log. May be we can print the userid and say passcode is persisted for this x user.